src/foam/nanos/approval/ApprovalRequest.js

```diff
@@ -772,6 +783,50 @@ foam.CLASS({
         }));
       }
     },
+    {
+      name: 'retry',
+      section: 'approvalRequestInformation',
+      isAvailable: async function(isTrackingRequest, status, subject, assignedTo, canRetry) {
+        if ( status !== this.ApprovalStatus.REQUESTED ) return false;
+        if ( assignedTo !== 0 && subject.realUser.id !== assignedTo ) return false;
+
+        return ! isTrackingRequest && canRetry;
+      },
+      code: async function(X) {
+        const approvalRequest = this.clone();
+
+        this.approvalRequestDAO.put(approvalRequest).then(req => {
+          this.approvalRequestDAO.cmd(foam.dao.DAO.RESET_CMD);
+          this.tableViewApprovalRequestDAO.cmd(foam.dao.DAO.RESET_CMD);
+          this.approvalRequestDAO.cmd(foam.dao.DAO.PURGE_CMD);
+          this.tableViewApprovalRequestDAO.cmd(foam.dao.DAO.PURGE_CMD);
+
+          this.finished.pub();
+          // Give some delay while approval request moves to the final state
+          setTimeout(() => {
```

Comment:
what happens if approval request finalization takes longer than 1sec or it doesnt get to final state?
is there a way we can wait for the correct status?